crates/oci/src/loader.rs

```diff
@@ -0,0 +1,127 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::{anyhow, ensure, Context, Result};
+use oci_distribution::Reference;
+use reqwest::Url;
+use spin_app::locked::{ContentPath, ContentRef, LockedApp, LockedComponent};
+use spin_loader::cache::Cache;
+
+use crate::{Client, ORIGIN_URL_SCHEME};
+
+/// OciLoader loads an OCI app in preparation for running with Spin.
+pub struct OciLoader {
+    working_dir: PathBuf,
+}
+
+impl OciLoader {
+    /// Creates a new OciLoader which builds temporary mount directory(s) in
+    /// the given working_dir.
+    pub fn new(working_dir: impl Into<PathBuf>) -> Self {
+        let working_dir = working_dir.into();
+        Self { working_dir }
+    }
+
+    /// Loads a LockedApp with the given OCI client and reference.
+    pub async fn load_app(&self, client: &mut Client, reference: &str) -> Result<LockedApp> {
+        // Fetch app
+        client
+            .pull(reference)
+            .await
+            .context("cannot pull Spin application from registry")?;
```

Comment:
It may seem like it should be obvious but I would like to include the reference in the context.  I am thinking of cases where we have incorrectly inferred a file path to be an OCI reference; rather than just telling the user "cannot pull Spin application" it might help to say "cannot pull x/y/z from registry" to prompt them to look at the string and ask "why is it trying to pull that" rather than "why is my application in a registry".  (Of course higher levels should help in the diagnosis too!)